Update and correction of French translation files (bitcoin_fr.ts,bitcoin_fr_CA.ts and bitcoin_fr_FR.ts)

thisiskeanyvy opened the pull request:
- Addition of all translations with understandable French
--> Update of the file translations (in bitcoin_fr.ts on lines 12;20;24;173;177;181;185;189;193;201;205;209;213;217;221;225;229;233;241;245;249;253;291;335;339;375;379;399;419;579;583;832;1000;1045;1187;1199;1370;1625;2236;2732;3030;3022;3034;3539;3561;3565;3573;3581;3628;3680;3684;3692;3704;3748;3780;3816;3848;3856;3860;3864;3964;4004;4008;4012;4016;4020;4036;4048;4080;4148;4204;4224;4324;4372;4432;4476 and 4480)
--> Update of the file translations (in bitcoin_fr_CA.ts on line 10)
--> Update of the file translations (in bitcoin_fr_FR.ts on lines 6;18;30;34;50;54;58;62;66;81;88;92;96;100;104;108;112;116;127;134;142;150;158;166;174;194;198;202;206;210;214;218;226;230;242;258;262;266;270;274;290;294;298;302;306;310;318;322;326;350;392;396;403;407;411;419;423;427;431;435;439;443;451;455;467;471;475;487;503;510;525;533;552;556;564;568;579;587;591;595;603;607;625;629;633;637;652;660;668;676;688;692;704;708;728;732;736;748;756;768;772;780;791;795;799;803;807;811;815;819;823;833;848;890;894;901;913;921;925;940;941;945;949;953;957;961;993;997;1041;1049;1053;1061;1132;1136;1148;1156;1171;1175;1179;1210;1225;1229;1233;1237;1245;1249;1253;1257;1265;1273;1289;1293;1301;1309;1325;1333;1352;1356;1364;1384;1401;1409;1417;1425;1433;1437;1441;1490;1498;1526;1560;1571;1595;1599;1607;1623;1650;1654;1670;1674;1678;1682;1699;1715;1719;1727;1731;1735;1739;1743;1747;1766;1767;1779;1783 and 1791)
- Deletion of type="unfinished" (on lines 619;781;981;1347;1351;1355;1359;1486;1490;1494;1498;1502;1506;1834;1838;1951;2324;2764;2972;4572 and 4576)
- Deletion of type="vanished" (on lines 303;626;765;949;1446;1450;1458;1470;1474;1478;1818;1935;1939;1947;2316;2740;3079;3111;3115;3282;3310 and 4540)
- _Correction of the grammar of the file (in bitcoin_en.ts on lines 1223;2509;2510;3999 and 5359)_

thisiskeanyvy commented:
Conflict resolution with the version 1.14.7-dev branch

thisiskeanyvy commented:
I'll do a commits squash after the review and when everything is ok.

thisiskeanyvy commented:
Everything is good for me.

Martyornot reviewed (approved):
Looks good to me. La traduction est parfaite! Good job @thisiskeanyvy!

thisiskeanyvy commented:
@Martyornot Merci beaucoup pour la review ! :)

sinetek commented:
Avons-nous besoin de la traduction fr_CA? Dans le dépot bitcoin d'origine c'est bitcoin_fr pour tout le monde.

thisiskeanyvy commented:
> Avons-nous besoin de la traduction fr_CA? Dans le dépot bitcoin d'origine c'est bitcoin_fr pour tout le monde.

La traduction Fr-ca était déjà présente dans le repo lors du fork de bitcoin pour créer Dogecoin. Au final la traduction est la même mais si des Canadiens trouvent des nuances pour certaines tournures de phrases ils pourront contribuer.

La traduction Fr-ca sert également dans le logiciel Dogecoin CORE pour changer de langue via l'interface Qt et il est important qu'elle y soit présente pour ne pas délaisser les Canadiens.

patricklodder commented:
Strings are now frozen so it's safe to finalize this. I have re-activated `fr_CA` as a maintained dialect. 

Could you please rebase this on 1.14.7-dev (anything after ad6a62b) so that this can go into final review?

thisiskeanyvy commented:
Sorry, I made a mistake, I stand corrected.

thisiskeanyvy commented:
> 1.14.7-dev

Normally all is well, I have corrected the error and squash all commits with a fetch and rebase on the latest version of 1.14.7-dev.

thisiskeanyvy commented:
@patricklodder Normally all is well, I have corrected the error and squash all commits with a fetch and rebase on the latest version of 1.14.7-dev

thisiskeanyvy commented:
@patricklodder Normally everything's fine, I've fixed the conflicts and squashed all the commits with a fetch and rebase on the latest version of 1.14.7-dev.

thisiskeanyvy commented:
@patricklodder Normally everything's fine, I've fixed the conflicts and squashed all the commits with a fetch and rebase on the latest version of 1.14.7-dev.

patricklodder commented on src/qt/locale/bitcoin_fr.ts:
```diff
@@ -1467,43 +1467,43 @@
     <message>
-        <source>Who owns the private keys owns the coins.</source>
-        <translation type="vanished">Ceux qui possèdent les clés privées possèdent les pièces.</translation>
+        <source>Who own the private keys own the coins.</source>
```
If we change the source text the translation will not show, so this needs to be reverted to the source from 1.14.7-dev

patricklodder reviewed (commented):
- bitcoin_en.ts should not be changed, because this is the base for ALL translations and is auto-generated. Please revert; can use `git checkout 07bc7361b5 -- src/qt/locale/bitcoin_en.ts` to do this from command line.
- bitcoin_fr_FR.ts was deprecated 20 months ago with #3011 and removed with #3413. If you're okay with letting that one go, please remove the `src/qt/locale/bitcoin_fr_FR.ts` file. If not, we can bring it back, but I'd like to understand what the added value is of `fr_FR` if we have a well-maintained `fr`. Note that I specifically kept `fr_CA` because every release people ask about it.

Not blocking: The "vanished" text can be deleted completely, right now it's just taking up space. Luckily, we can automate that with a bit of `lupdate` trickery. I propose that we just run that in a separate cleanup PR once this is done, and not bother going through hundreds of entries manually right now.

thisiskeanyvy commented:
Thanks a lot for the help ! :)

patricklodder reviewed (commented):
If you could squash these 2 commits into one, I think this is good to go, especially because language was validated prior.

I'll propose an cleanup PR from an automated script in a separate PR on top of this then.

patricklodder reviewed (approved):
ACK - code review OK

Language validated by @Martyornot
